cardano-api/src/Cardano/API.hs

```diff
@@ -0,0 +1,265 @@
+    SerialiseAsRawBytes,
+    serialiseToRawBytes,
+    deserialiseFromRawBytes,
+    serialiseToRawBytesHex,
+    deserialiseFromRawBytesHex,
+
+    -- ** Text envelope
+    -- | Support for a envelope file format with text headers and a hex-encoded
+    -- binary payload.
+    HasTextEnvelope,
+    TextEnvelope,
+    TextEnvelopeType,
+    TextEnvelopeDescr,
+    TextEnvelopeError,
+    serialiseToTextEnvelope,
+    deserialiseFromTextEnvelope,
+    readFileTextEnvelope,
+    writeFileTextEnvelope,
+    readTextEnvelopeFromFile,
+    readTextEnvelopeOfTypeFromFile,
+    -- *** Reading one of several key types
+    FromSomeType,
+    deserialiseFromTextEnvelopeAnyOf,
+    readFileTextEnvelopeAnyOf,
+
+    -- * Errors
+    Error,
+    throwErrorAsException,
+    FileError,
+
+    -- * Node interaction
+    -- | Operations that involve talking to a local Cardano node.
+
+    -- ** Queries
+    -- ** Submitting transactions
+
+    -- ** Low level protocol interaction with a Cardano node
+    connectToLocalNode,
+    LocalNodeConnectInfo,
+    localNodeSocketPath,
+    localNodeNetworkId,
+    localNodeConsensusMode,
+    NodeConsensusMode,
+    LocalNodeClientProtocols,
+    localChainSyncClient,
+    localTxSubmissionClient,
+    localStateQueryClient,
+    nullLocalNodeClientProtocols,
+--  connectToRemoteNode,
+
+    -- *** Chain sync protocol
+    ChainSyncClient,
+    runChainSyncClient,
+
+    -- *** Local tx submission
+    LocalTxSubmissionClient,
+    runLocalTxSubmissionClient,
+    submitTxToNodeLocal,
+
+    -- *** Local state query
+    LocalStateQueryClient,
+    runLocalStateQueryClient,
+    queryNodeLocalState,
+
+    -- * Node operation
+    -- | Support for the steps needed to operate a node
+
+    -- ** Operational certificates
+    OperationalCertificate,
+    OperationalCertificateIssueCounter,
+    OperationalCertIssueError,
+    issueOperationalCertificate,
+
+    -- * Genesis file
+    -- | Types and functions needed to inspect or create a genesis file.
+    GenesisKey,
+    GenesisExtendedKey,
+    GenesisDelegateKey,
+    GenesisDelegateExtendedKey,
+    GenesisUTxOKey,
+    genesisUTxOPseudoTxIn,
+
+    -- * Special transactions
+    -- | There are various additional things that can be embedded in a
+    -- transaction for special operations.
+    makeMIRCertificate,
+    makeGenesisKeyDelegationCertificate,
+
+    NetworkMagic,
+    makeShelleyUpdateProposal,
+  ) where
+
+-- | This module provides a library interface for interacting with Cardano as
+-- a user of the system.
+--
+-- It is intended to be the complete API covering everything but without exposing
+-- constructors that reveal any lower level types.
+--
+-- In the interest of simplicity it glosses over some details of the system.
+-- Most simple tools should be able to work just using this interface,
+-- however you can go deeper and expose the types from the underlying libraries
+-- using "Cardano.Api.Byron" or "Cardano.Api.Shelley".
+--
```

Comment:
Does putting the haddock comment here work? I thought it needed to be before the `module` declaration.